oxo.py

```diff
@@ -1,26 +1,101 @@
 class OxoBoard:
-    def __init__(self):
+    def __init__(self, custom_width=3, custom_height=3):
         """ The initialiser. Initialise any fields you need here. """
-        raise NotImplementedError("TODO: implement __init__")
+
+        self.board = {}
+        self.width = custom_width
+        self.height = custom_height
+
+        for x in range(0, custom_width):
+            for y in range(0, custom_height):
+                self.board[x, y] = 0
 
     def get_square(self, x, y):
         """ Return 0, 1 or 2 depending on the contents of the specified square. """
-        raise NotImplementedError("TODO: implement get_square")
+        return self.board[x, y]
 
     def set_square(self, x, y, mark):
         """ If the specified square is currently empty (0), fill it with mark and return True.
             If the square is not empty, leave it as-is and return False. """
-        raise NotImplementedError("TODO: implement set_square")
+
+        if self.board[x, y] == 0:
+            self.board[x, y] = mark
+            return True
+        else:
+            return False
 
     def is_board_full(self):
         """ If there are still empty squares on the board, return False.
             If there are no empty squares, return True. """
-        raise NotImplementedError("TODO: implement is_board_full")
 
-    def get_winner(self):
+        return False if 0 in self.board.values() else True
+
+    def get_winner(self, row_length=3):
         """ If a player has three in a row, return 1 or 2 depending on which player.
             Otherwise, return 0. """
-        raise NotImplementedError("TODO: implement get_winner")
+
+        for cell in self.board:
+
+            if self.board[cell] != 0:
```

Comment:
It would be slightly more efficient to use `items` to avoid having to look up `self.board[cell]`:
```
for cell, cell_contents in self.board.items():
```
Or even, since you are using `cell[0]` and `cell[1]` a lot:
```
for (cell_x, cell_y), cell_contents in self.board.items():
```